src/states/ContentPlayback.js

```diff
@@ -7,16 +8,44 @@ import {ContentState, Midroll, Postroll} from '../states.js';
  */
 export default class ContentPlayback extends ContentState {
 
-  init(player) {
-    // Play the content if cancelContentPlay happened or we paused on 'contentupdate'
+  init(player, resumingAfterNoPreroll) {
+    this.resumingAfterNoPreroll = resumingAfterNoPreroll || false;
+
+    // eslint-disable-next-line no-console
+    console.log('*$*', '_playRequested', player.ads._playRequested,
+      '_shouldBlockPlay', player.ads._shouldBlockPlay,
+      '_cancelledPlay', player.ads._cancelledPlay,
+      '_pausedOnContentupdate', player.ads._pausedOnContentupdate);
+
+    // Don't block calls to play in content playback
+    player.ads._shouldBlockPlay = false;
+
+    // Play the content if we had requested play or we paused on 'contentupdate'
     // and we haven't played yet. This happens if there was no preroll or if it
     // errored, timed out, etc. Otherwise snapshot restore would play.
     if (player.paused() &&
-        (player.ads._cancelledPlay || player.ads._pausedOnContentupdate)) {
+        (player.ads._playRequested || player.ads._pausedOnContentupdate)) {
+      // TODO: test snapshot restores
+      // TODO: test playlists
       player.play();
     }
   }
 
+  /*
+   * When there is no preroll, we immediately enter ContentPlayback
+   * State, but we are still resuming to content.
+   */
+  isResumingAfterNoPreroll() {
+    return this.resumingAfterNoPreroll;
+  }
+
+  /*
+   * Reset resumingAfterNoPreroll once the content actually starts playing
+   */
+  onPlaying(player) {
+    this.resumingAfterNoPreroll = false;
```

Comment:
I believe we transition to Content Mode after a `playing` event, so would this be the second playing event? Not sure that resumingAfterNoPreroll needs to be part of ContentPlayback and that the ad states could handle it.

Reply:
In that case, would it work to set `this.resumingAfterNoPreroll = false;` in `ContentPlayback.init`?

Reply:
Ah, I think this further indicates an issue with how we're handling nopreroll. Normally we transition to ContentPlayback after playing, but in this case we don't. As a result, preroll logic is creeping into ContentPlayback.

What do you think of setting resumingAfterNoPreroll within Preroll, and transitioning to Preroll on nopreroll (heh) with resumingAfterNoPreroll and contentResuming both set to true?

Reply:
so `Preroll -> Preroll`? 

Honestly I think this would be a big change and wouldn't fit in the context of this PR. I think making resumingAfterNoPreroll "first-class" should be done separately, and really we shouldn't have the special resumingAfterNoPreroll flag and instead go into contentResuming until `playing`.

Reply:
I agree with avoiding the special `resumingAfterNoPreroll` flag and staying in Preroll until `playing`. The state transition would be `BeforePreroll -> Preroll` if `nopreroll` happened earlier. If `nopreroll` happened during `Preroll` it would stay in that state and set `contentResuming` to `true`.

The reason I think it fits into this PR is that you're creating technical debt (the preroll handling in ContentPlayback) to work around the problem. Rather than introduce new technical debt with the intent to remove it later, it's preferable to not introduce it at all.

I'm 100% down to assist in making this change.